### ISSUE TITLE
`ScrollView` scrollbar actions don't trigger `scrolled` callback

### ISSUE THREAD

**Issue by bryce-happel-walton** [slint-ui/slint#8170]
### Bug Description

Any interaction with the scrollbar does not call the `scrolled` callback.

### Reproducible Code (if applicable)

```slint
ScrollView {
 scrolled => {
  ...
 }

 vertical-scrollbar-policy: ScrollBarPolicy.always-on;
}
```

### Environment Details

- Slint Version: 1.11.0 git 691039b7ea76cd5e238de2128db1d4654a51c9ae
- Platform/OS: Fedora Linux 41 Kernel 6.13.10-200.fc41.x86_64 (64-bit)
- Programming Language: Slint
- Backend/Renderer: Preview/Skia

**Comment by ogoffart** [comment]
I can reproduce.
Scrolling with the wheel works, but changing the value of the scrollbar doesn't

Here is a self-contained code to reproduce.

```slint
import {  ScrollView } from "std-widgets.slint";

export component Demo {
    ScrollView {
        HorizontalLayout {
            Rectangle {
                width: 3000px;
                height: 3000px;
                background: @radial-gradient(circle, #6823d4ff 0.00%, #d46423ff 100.00%);
            }
        }
        scrolled => {
            debug("scrolled", self.viewport-y, self.viewport-x)
        }
        vertical-scrollbar-policy: ScrollBarPolicy.always-on;
    }
}
```
